### ISSUE TITLE
Magento 2 and SSL connection to MySQL

### ISSUE THREAD

**Issue by cscd98** [magento/magento2#13561]
### Preconditions
1. Magento 2.2.2 
2. MySQL 5.7.x

### Steps to reproduce
1. In env.php there should be a way to specify optional secure connection details which can then be used: 

```
$this->resource->ssl_set($clientKey, $clientCert, $caCert, $caPath, $cipher);
```

To establish a secure connection to mysql.

### Expected result
1. Secure connection

### Actual result
1. I don't believe this is supported?

Example of what I mean:
https://stackoverflow.com/questions/9738712/connect-to-remote-mysql-server-with-ssl-from-php
http://blog.machek.co.uk/2016/06/php-with-mysql-and-ssl.html

**Comment by magento-engcom-team** [comment]
@craigcarnell, thank you for your report.
This seems to be correct Magento behavior. Please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/) site for advice or general discussion about this.
Otherwise you may submit Pull Request with the suggested changes.

**Comment by bnymn** [comment]
Hi @craigcarnell,

You can pass your SSL connection configurations inside of the **env.php** as follows.

![screenshot from 2018-04-04 08-38-58](https://user-images.githubusercontent.com/3506402/38305870-dc01f580-37e4-11e8-97e8-52289fe6ef03.png)

For more information about driver_options parameter please follow [this link](http://php.net/manual/en/ref.pdo-mysql.php#103501).

**Comment by magento-engcom-team** [comment]
Hi @craigcarnell. Thank you for the report. We are moving your feature request to the special project. You can track this issue here: https://github.com/magento/community-features/issues/61

**Comment by magento-engcom-team** [comment]
Hi @craigcarnell. Thank you for your report. 
The issue has been fixed in magento/magento2#18075 by @bnymn in 2.3-develop branch
Related commit(s):
 - [41e17938b6b3b6ea94e0cb0552727eb596341782](https://github.com/magento/magento2/commit/41e17938b6b3b6ea94e0cb0552727eb596341782)
 - [3b67548c8c85686dc6ae34995f84f412e844d2af](https://github.com/magento/magento2/commit/3b67548c8c85686dc6ae34995f84f412e844d2af)
 - [def50d54c2246e3ae265554dc581e39ac8fcd004](https://github.com/magento/magento2/commit/def50d54c2246e3ae265554dc581e39ac8fcd004)
 - [7c7cc9bb16ecf5deaeec3b334f0443b058432133](https://github.com/magento/magento2/commit/7c7cc9bb16ecf5deaeec3b334f0443b058432133)
 - [ddad680b699418ed9941359df7540eb8120abc79](https://github.com/magento/magento2/commit/ddad680b699418ed9941359df7540eb8120abc79)
 - [fa659d20695b9cbfe7b4f4ab33d1ee35673baa55](https://github.com/magento/magento2/commit/fa659d20695b9cbfe7b4f4ab33d1ee35673baa55)
 - [968a59ccf2a079d21b2c61fee008132cdc398b01](https://github.com/magento/magento2/commit/968a59ccf2a079d21b2c61fee008132cdc398b01)
 - [5cfad033e00338456f6eed9da68e3632dfaeaad9](https://github.com/magento/magento2/commit/5cfad033e00338456f6eed9da68e3632dfaeaad9)
 - [7c7760ba30ac87eabceee22a13b40c11c4bb60a0](https://github.com/magento/magento2/commit/7c7760ba30ac87eabceee22a13b40c11c4bb60a0)
 - [6f19bf097b321356ffec4798f42d457c5f743256](https://github.com/magento/magento2/commit/6f19bf097b321356ffec4798f42d457c5f743256)
 - [d97af677ad0b602aa5f1bb697ef5360c6d324406](https://github.com/magento/magento2/commit/d97af677ad0b602aa5f1bb697ef5360c6d324406)
 - [6cd243d96bc1ea2116bd9cd28aae4c83f828480e](https://github.com/magento/magento2/commit/6cd243d96bc1ea2116bd9cd28aae4c83f828480e)
 - [7b137053ad15f71942fa8f699231e812cbb00137](https://github.com/magento/magento2/commit/7b137053ad15f71942fa8f699231e812cbb00137)
 - [93931d2c8359c9da14babeb7d76593007127a7a2](https://github.com/magento/magento2/commit/93931d2c8359c9da14babeb7d76593007127a7a2)
 - [db60f716713815ae6265855647146027f4eef965](https://github.com/magento/magento2/commit/db60f716713815ae6265855647146027f4eef965)
 - [d27e9a4e885b1480b6835d01f4e17fa0ef455cc9](https://github.com/magento/magento2/commit/d27e9a4e885b1480b6835d01f4e17fa0ef455cc9)
 - [f06a4b74840de2a7c6697787e27be48d43e5fb90](https://github.com/magento/magento2/commit/f06a4b74840de2a7c6697787e27be48d43e5fb90)
 - [309e906657d2fa8cbc9f8f20bde2be257340a482](https://github.com/magento/magento2/commit/309e906657d2fa8cbc9f8f20bde2be257340a482)
 - [f04a5a9302c1d188d34275c3635bdbd33668664d](https://github.com/magento/magento2/commit/f04a5a9302c1d188d34275c3635bdbd33668664d)
 - [f7ee0cc4e333d84e7f683691a52dc73d3a4376e6](https://github.com/magento/magento2/commit/f7ee0cc4e333d84e7f683691a52dc73d3a4376e6)
 - [c4cc9523a34b561d6768eb00b23800c5888d8a1e](https://github.com/magento/magento2/commit/c4cc9523a34b561d6768eb00b23800c5888d8a1e)
 - [26980d84167f6f1c282c9dbf45554215e592ae30](https://github.com/magento/magento2/commit/26980d84167f6f1c282c9dbf45554215e592ae30)
 - [610dcff530b79ae89c3bb7ac557b4deec49c4d99](https://github.com/magento/magento2/commit/610dcff530b79ae89c3bb7ac557b4deec49c4d99)

The fix will be available with the upcoming 2.3.3 release.
